### DISCUSSION TITLE
what is the difference between `unlistedscript` and `content script`?

### DISCUSSION THREAD

**Discussion by xunziheng** [wxt-dev/wxt#702] [Q&A] [answered]
Could you tell me the difference between `unlistedscript` and `content script`? If I want to manually inject content script based on URL, how can I do it? (Matches can't satisfy me.)

Regards, wxt

for example: 

```js
export default defineBackground({
  main() {
    console.log('Hello background!', { id: browser.runtime.id });

    browser.tabs.onUpdated.addListener(async (tabId, changeInfo, tab) => {
      if (status === 'complete' && isTestPage(url)) {
        await (browser.action ?? browser.browserAction).enable(tabId);
        // >>>>>>>TODO: how to inject content script? (main world needed)
      } else {
        await (browser.action ?? browser.browserAction).disable(tabId);
      }
    });
  },
});
```

**Comment by aklinker1** [comment, marked as answer]
Unlisted scripts aren't added to the manifest. Content scripts are listed in the manifest and are automatically registered to run in tabs.

You can use the `scripting` API to run code in a tab whenever you want: https://developer.chrome.com/docs/extensions/reference/api/scripting#concepts_and_usage

**Reply by xunziheng** [reply]
I saw that `browser.scripting.executeScript()` only support inject isolated world script right.
but the chrome extension guide show 2 worlds:
<img width="669" alt="image" src="https://github.com/wxt-dev/wxt/assets/35691153/7c4efafd-ff56-4e5d-86ce-7d0651909467">
<img width="924" alt="image" src="https://github.com/wxt-dev/wxt/assets/35691153/d3cbac93-17d4-445a-b1d0-3b33a57250fc">

**Reply by aklinker1** [reply]
> I saw that browser.scripting.executeScript() only support inject isolated world script right.

That's not true. You can execute the script in the main world.

https://developer.chrome.com/docs/extensions/reference/api/scripting#type-ScriptInjection

**Reply by xunziheng** [reply]
I'll try it, thanks aklinker1.
